cpp/unreal_projects/SpearSim/Source/SpearSim/SpearSimSpectatorPawn.cpp

```diff
@@ -4,43 +4,41 @@
 
 #include "SpearSim/SpearSimSpectatorPawn.h"
 
-#include <Components/InputComponent.h>
-#include <GameFramework/PlayerInput.h>
+#include <string>
+
 #include <GameFramework/SpectatorPawn.h>
 #include <GenericPlatform/GenericPlatformMisc.h>
 
-#include "CoreUtils/Config.h"
+#include "CoreUtils/InputActionComponent.h"
 #include "CoreUtils/Unreal.h"
 
-class FObjectInitializer;
-
-ASpearSimSpectatorPawn::ASpearSimSpectatorPawn(const FObjectInitializer& object_initializer) : ASpectatorPawn(object_initializer)
+ASpearSimSpectatorPawn::ASpearSimSpectatorPawn()
 {
     SP_LOG_CURRENT_FUNCTION();
 
-    if (!Config::s_initialized_) {
-        return;
-    }
+    // UInputActionComponent
+    input_action_component_ = CreateDefaultSubobject<UInputActionComponent>(Unreal::toFName("input_action_component"));
+    SP_ASSERT(input_action_component_);
+    // Need to explicitly set this up so that the component hierarchy is well-defined.
+    input_action_component_->SetupAttachment(GetRootComponent());
```

Comment:
I think we should call RegisterComponent after SetupAttachment because we do that in other places in the code. Is this situation different?

Reply:
When we create components using CreateDefaultSubobject (in the constructor), we don't need to call RegisterComponent

Reply:
Got it.